File: ContactsManagerAppWithJavaROOMDBMVVM/app/src/main/java/com/masterclass/contactsmanagerappwithjavaroomdbmvvm/adapters/ContactsAdapter.java
package com.masterclass.contactsmanagerappwithjavaroomdbmvvm.adapters;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.R;
import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.database.Contacts;
import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.databinding.ContactListItemBinding;

import java.util.ArrayList;

public class ContactsAdapter extends RecyclerView.Adapter<ContactsAdapter.ContactsViewHolder> {

    private ArrayList<Contacts> contacts;

    public ContactsAdapter(ArrayList<Contacts> contacts) {
        this.contacts = contacts;
    }

    @NonNull
    @Override
    public ContactsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        ContactListItemBinding contactListItemBinding = DataBindingUtil.inflate(
                LayoutInflater.from(parent.getContext()),
                R.layout.contact_list_item,
                parent,
                false
        );

        return new ContactsViewHolder(contactListItemBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ContactsViewHolder holder, int position) {
        Contacts currentContact = contacts.get(position);
        holder.contactListItemBinding.setContact(currentContact);
    }

    @Override
    public int getItemCount() {
        if (contacts != null){
            return contacts.size();
        }else {
            return 0;
        }
    }

    @SuppressLint("NotifyDataSetChanged")
    public void setContacts(ArrayList<Contacts> contacts) {
        this.contacts = contacts;

        notifyDataSetChanged();
    }

    static class ContactsViewHolder extends RecyclerView.ViewHolder{
        private final ContactListItemBinding contactListItemBinding;

        public ContactsViewHolder(@NonNull ContactListItemBinding contactListItemBinding) {
            super(contactListItemBinding.getRoot());
            this.contactListItemBinding = contactListItemBinding;
        }
    }
}

File: ContactsManagerAppWithKotlinROOMDBMVVM2/app/src/main/java/com/masterclass/contactsmanagerappwithkotlinroomdbmvvm/adapters/ContactsAdapter.kt
package com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.adapters

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.R
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.database.Contacts
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.databinding.ContactListItemBinding


class ContactsAdapter(private var contacts: ArrayList<Contacts>) :
    RecyclerView.Adapter<ContactsAdapter.ContactsViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContactsViewHolder {
        val contactListItemBinding = DataBindingUtil.inflate<ContactListItemBinding>(
            LayoutInflater.from(parent.context),
            R.layout.contact_list_item,
            parent,
            false
        )
        return ContactsViewHolder(contactListItemBinding)
    }

    override fun onBindViewHolder(holder: ContactsViewHolder, position: Int) {
        val currentContact = contacts[position]
        holder.contactListItemBinding.contact = currentContact
    }

    override fun getItemCount(): Int {
        return contacts?.size ?: 0
    }

    @SuppressLint("NotifyDataSetChanged")
    fun setContacts(contacts: ArrayList<Contacts>) {
        this.contacts = contacts
        notifyDataSetChanged()
    }

    class ContactsViewHolder(val contactListItemBinding: ContactListItemBinding) :
        RecyclerView.ViewHolder(contactListItemBinding.root)
}